### PR TITLE
[None][feat] Add support for DeepSeek-V4 in PyTorch backend

### LINKED ISSUE #13435: [Feat] Support DeepSeek-V4 in PyTorch backend
This issue tracks the support for DeepSeek-V4 in the PyTorch backend. A PR is already created at https://github.com/NVIDIA/TensorRT-LLM/pull/13434.

### PR DESCRIPTION
**Pull request by weikuo0506**
This PR adds support for DeepSeek-V4 in the PyTorch backend.

**Key Verifications:**
*   **Dummy forward pass** and **decoding loop tests** pass successfully.
*   **Logits match the official implementation exactly** (**Max absolute difference: 0.0**) on small dummy models matching both **DeepSeek-V4-Flash** and **DeepSeek-V4-Pro** configurations.

Fixes #13435

### CODE REVIEW

**Comment by weikuo0506** [comment]
@coderabbitai review

**Comment by juney-nvidia** [comment]
Thanks for the contributions, @weikuo0506 !

Just making sure you are aware of the ongoing efforts from TensorRT LLM team of adding the DS V4 support which currently is on a feature branch(we plan to trigger the process of landing it to the main branch recently):

- https://github.com/NVIDIA/TensorRT-LLM/tree/feat/deepseek_v4

Here is the user guide:

- https://github.com/NVIDIA/TensorRT-LLM/blob/feat/deepseek_v4/examples/models/core/deepseek_v4/README.md

And here is the current support matrix

- https://github.com/NVIDIA/TensorRT-LLM/blob/feat/deepseek_v4/docs/source/models/supported-models.md

Since the changes on the feature branch is pretty huge, it may not be easy to directly land your PRs to the GH main branch, while if you have interest, I think we can still discuss to figure out how to let you get engaged into TRTLLM contributions.

Thanks
June

**Comment by weikuo0506** [comment]
Hi June @juney-nvidia,

Thanks for the heads up and for sharing the links! 

I've taken a look at the `feat/deepseek_v4` branch, and the official implementation is very comprehensive. I will close this PR to avoid duplication and keep efforts aligned with the official roadmap. 

If there is anything I can help with down the road, I'd be more than happy to contribute. Looking forward to seeing DeepSeek-V4 support land on the main branch!

Best regards,
Kuo
